### PR TITLE
Fix SecurityTest for back compatibility tests and fix skip logic

### PR DESCRIPTION
**Pull request by srknzl**
Origin of PR by Ali Sengul's compat test run: https://github.com/hazelcast/client-compatibility-suites/runs/4773613227?check_suite_focus=true

A broken link fix was also necessary.

### CODE REVIEW

File: test/integration/backward_compatible/parallel/security/SecurityTest.js [hunk 1]
```diff
@@ -84,14 +100,24 @@ describe('SecurityTest', function () {
         let cluster;
         let member;
         const testFactory = new TestUtil.TestFactory();
+        let tokenEncoding;
 
         before(async function () {
             TestUtil.markEnterprise(this);
-            TestUtil.markClientVersionAtLeast(this, '5.1');
+            // Security config is added in 5.1.0, 4.2.1 and 5.0.3.
+            // Security config is added in 5.1.0, 4.2.1 and 5.0.3.
```
Comment (yuce) [inline test/integration/backward_compatible/parallel/security/SecurityTest.js:108]: I guess no need for the duplicated comment?

File: test/integration/backward_compatible/parallel/security/SecurityTest.js [hunk 2]
```diff
@@ -137,14 +163,24 @@ describe('SecurityTest', function () {
         let cluster;
         let member;
         const testFactory = new TestUtil.TestFactory();
+        let simpleCredentials;
 
         before(async function () {
             TestUtil.markEnterprise(this);
-            TestUtil.markClientVersionAtLeast(this, '5.1');
+            // Security config is added in 5.1.0, 4.2.1 and 5.0.3.
+            // Security config is added in 5.1.0, 4.2.1 and 5.0.3.
```
Comment (yuce) [inline test/integration/backward_compatible/parallel/security/SecurityTest.js:171]: Duplicated comment here too.

**Review by yuce — APPROVED** [review]
Looks great!

File: test/integration/backward_compatible/parallel/security/SecurityTest.js (hunk 1 above)
Comment (srknzl) [reply test/integration/backward_compatible/parallel/security/SecurityTest.js]: ok there is another one below. Fixed both at bb84c238

File: test/integration/backward_compatible/parallel/security/SecurityTest.js [hunk 3]
```diff
@@ -23,8 +23,17 @@ const TestUtil = require('../../../../TestUtil');
...
+// Lazy imports are due to back compatibility
+const getTokenEncoding = () => {
+    const {TokenEncoding} = require('../../../../../lib/security/TokenEncoding');
+    return TokenEncoding;
+};
+
+const getSimpleCredentials = () => {
+    const {SimpleCredentials} = require('./SimpleCredentials');
+    return SimpleCredentials;
+};
```
Comment (mdumandag) [inline test/integration/backward_compatible/parallel/security/SecurityTest.js:36]: This is not necessary, as it is a custom file that sits in the tests directory

Comment (srknzl) [reply test/integration/backward_compatible/parallel/security/SecurityTest.js]: Fixed at f93de860

**Review by mdumandag — APPROVED** [review]

**Pull request merged**